src/bin/pg_autoctl/keeper_pg_init.c

```diff
@@ -86,6 +88,198 @@ keeper_pg_init_fsm(Keeper *keeper, KeeperConfig *config)
 }
 
 
+/*
+ * keeper_ensure_pg_configuration_files_in_pgdata checks if
+ * postgresql.conf, pg_hba.conf, pg_ident.conf files exist
+ * in $PGDATA, if not it tries to get them from default
+ * location and modifies paths inside copied postgresql.conf.
+ */
+static bool
+keeper_ensure_pg_configuration_files_in_pgdata(KeeperConfig *config)
+{
+	char pgdata[MAXPGPATH];
+	char hbaConfPath[MAXPGPATH];
+	char posgresqlConfPath[MAXPGPATH];
+	char *pgdataRelativePath = NULL;
+	char defaultConfigurationPath[MAXPGPATH];
+	bool configurationPathPresent = false;
+	bool postgresqlConfExists = false;
+	char defaultPosgresqlConfPath[MAXPGPATH];
+	char defaultHbaConfPath[MAXPGPATH];
+	char defaultIdentConfPath[MAXPGPATH];
+	char identConfPath[MAXPGPATH];
+
+	get_absolute_path(config->pgSetup.pgdata, pgdata, MAXPGPATH);
+
+	/* check if postgresql.conf exists in pgdata. there is no separate check for pg_hba and pg_ident  */
+	join_path_components(posgresqlConfPath, pgdata, "postgresql.conf");
+	postgresqlConfExists = file_exists(posgresqlConfPath);
+	if (postgresqlConfExists)
+	{
+		return true;
+	}
+
+	/* check if pgdata is in default path. */
+	if (strstr(pgdata, "/var/lib/postgresql") != pgdata)
```

Comment:
What do we mean by "default path" here? I guess you mean debian packaging default path? but that's very specific to debian style packaging, right? We could maybe `#define` a constant `DEBIAN_PACKAGING_DEFAULT_PGDATA_TOP_DIRECTORY` or something?